cargo-dist/templates/installer/installer.sh.j2

```diff
@@ -287,93 +290,84 @@ install() {
     # Potentially-late-bound version of env_script_path to write to rcfiles like $HOME/.profile
     local _env_script_path_expr
 
-{% if install_path.kind == "CargoHome" %}
-    # first try CARGO_HOME, then fallback to HOME
-    if [ -n "${CARGO_HOME:-}" ]; then
-        _install_dir="$CARGO_HOME/bin"
-        _env_script_path="$CARGO_HOME/env"
-
-        # If we need to override the above
-        if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ] && [ "${CARGO_DIST_FORCE_INSTALL_DIR}" != "$CARGO_HOME" ]; then
-            _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR/bin"
-            _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
-        fi
-
-        # If CARGO_HOME was set but it ended up being the default $HOME-based path,
-        # then keep things late-bound. Otherwise bake the value for safety.
-        # This is what rustup does, and accurately reproducing it is useful.
-        if [ -n "${HOME:-}" ]; then
-            if [ "$HOME/.cargo/bin" = "$_install_dir" ]; then
+    # Before actually consulting the configured install strategy, see
+    # if we're overriding it.
+    if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ]; then
+        {% if install_paths| selectattr("kind", "equalto", "CargoHome") %}
+        _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR/bin"
+        _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
+        _install_dir_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/bin" | sed "s,$HOME,\$HOME,")"
+        _env_script_path_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/env" | sed "s,$HOME,\$HOME,")"
+        {%- else -%}
+        _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR"
+        _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
+        # For these two, if the new value contains $HOME, make
+        # sure to `sed` the $HOME expression back into the expr
+        _install_dir_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR" | sed "s,$HOME,\$HOME,")"
+        _env_script_path_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/env" | sed "s,$HOME,\$HOME,")"
+        {% endif %}
+    # If we're *not* overriding, go ahead with the more complex
+    # path evaluation logic
+    else
+    {% for install_path in install_paths %}
+        if [ -z "${_install_dir:-}" ]; then
```

Comment:
you don't need this else, it's implicit in the fact that `CARGO_DIST_FORCE_INSTALL_DIR` will set `_install_dir`, which nicely makes all the options look like a clean cascade :)

Reply:
This isn't trying to catch it being unset from the above case. We're generating an `if []` for each of the `for install_path in install_paths` cases, so it's letting item 2 in the array check if it was assigned in case 1, and so on.

Reply:
yes, i'm just saying we can omit the else because every case has this if-guard, making the whole thing feel a lot more natural (i applied this change)